src/lib/Libattr/attr_get_set.c

```diff
@@ -0,0 +1,290 @@
+	if ((rc = pdef->at_decode(&tempat, pdef->at_name, NULL, value)) != 0) {
+		log_errf(rc, __func__, "decode of %s failed", pdef->at_name);
+		return 1;
+	}
+
+	rc = set_attr_with_attr(pdef, pattr, &tempat, op);
+
+	pdef->at_free(&tempat);
+
+	return rc;
+}
+
+/**
+ * @brief	Set attribute using another attribute
+ *
+ * @param[in]	pdef 	-	attribute definition
+ * @param[in]	oattr	-	pointer to attribute being set
+ * @param[in]	nattr	-	pointer to attribute to set with
+ * @param[in]	op		-	operation to do
+ *
+ * @return	int
+ * @retval	0 for success
+ * @retval	1 for failure
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ *
+ */
+int
+set_attr_with_attr(attribute_def *pdef, attribute *oattr, attribute *nattr, enum batch_op op)
+{
+	int rc;
+
+	if ((rc = pdef->at_set(oattr, nattr, op)) != 0)
+		log_errf(rc, __func__, "set of %s failed", pdef->at_name);
+
+	return rc;
+}
+
+/**
+ * @brief	Attribute setter function for long type values
+ *
+ * @param[in]	pattr	-	pointer to attribute being set
+ * @param[in]	value	-	value to be set
+ *
+ * @return	void
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ *
+ */
+void
+set_attr_l(attribute *pattr, long value, enum batch_op op)
+{
+	if (pattr == NULL) {
+		log_err(-1, __func__, "Invalid pointer to attribute");
+		return;
+	}
+
+	switch (op) {
+		case SET:
+			pattr->at_val.at_long = value;
+			break;
+		case INCR:
+			pattr->at_val.at_long += value;
+			break;
+		case DECR:
+			pattr->at_val.at_long -= value;
+			break;
+		default:
+			return;
+	}
+
+	pattr->at_flags |= ATR_SET_MOD_MCACHE;
+}
+
+/**
+ * @brief	Attribute setter function for char type values
+ *
+ * @param[in]	pattr	-	pointer to attribute being set
+ * @param[in]	value	-	value to be set
+ *
+ * @return	void
+ *
+ * @par MT-Safe: No
+ * @par Side Effects: None
+ *
+ */
+void
+set_attr_c(attribute *pattr, char value, enum batch_op op)
+{
+	if (pattr == NULL) {
+		log_err(-1, __func__, "Invalid pointer to attribute");
+		return;
+	}
+
+	switch (op) {
+		case SET:
+			pattr->at_val.at_char = value;
+			break;
+		case INCR:
+			pattr->at_val.at_char += value;
+			break;
+		case DECR:
+			pattr->at_val.at_char -= value;
+			break;
+		default:
+			return;
+	}
+
+	pattr->at_flags |= ATR_SET_MOD_MCACHE;
+}
+
+void
+set_attr_b(attribute *pattr, long val, enum batch_op op)
+{
+	switch (op) {
+		case SET:
+			pattr->at_val.at_long = val;
+			break;
+
+		case INCR:
+			pattr->at_val.at_long = pattr->at_val.at_long | val; /* "or" */
+			break;
+
+		case DECR:
+			pattr->at_val.at_long = pattr->at_val.at_long & ~val;
+			break;
+
+		default:
+			return;
+	}
+	pattr->at_flags |= ATR_SET_MOD_MCACHE;
+}
+
+void
+set_attr_str(attribute *pattr, char *value)
+{
+	if (pattr == NULL) {
+		log_err(-1, __func__, "Invalid pointer to attribute");
+		return;
+	}
+
+	if (pattr->at_val.at_str != NULL)
+		free(pattr->at_val.at_str);
+
+	if (value == NULL)
+		pattr->at_val.at_str = NULL;
+	else
+		pattr->at_val.at_str = strdup(value);
```

Comment:
why don't we set attribute flags here?

Reply:
this function is actually a dud, it's not called by anything, I realized later that the code calls at_decode() for setting string type attirbutes, and some of them can do things more than just setting the attribute, so I decided to use set_attr_generic() for setting strings. So, I'll just remove this function.